### DISCUSSION TITLE
Setting Material Parameters and Crystal Plasticity Models with EBSD in CPFEM

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#26358] [Q&A Modules: Solid mechanics]
Dear Moose experts,

I hope this message finds you well. I am seeking technical guidance on how to set up the material parameters and crystal plasticity model for phase 1 and phase 2 respectively, when CPFEM uses the initial microstructure from EBSD within the MOOSE framework.

Primarily, I have two questions:

1. Is it possible to use EBSD data as the initial microstructure for CPFEM simulations in MOOSE?
2. How can I set different material parameters and Crystal Plasticity (CP) models for phase 1 and phase 2 after utilizing EBSD as the initial microstructure?

If there exists a well-established technical solution to accomplish these objectives, I would greatly appreciate guidance on the specific steps to follow. However, if such a solution is not readily available, I am interested in exploring the creation of a custom C++ class to achieve this purpose.

My aim is to effectively incorporate EBSD-derived microstructures into CPFEM simulations and establish distinct material parameters and CP models for different phases within the simulation.

Any insights, suggestions, or guidance on established methods or custom class creation within the MOOSE framework would be immensely helpful in addressing these challenges.

Thank you in advance for your time and expertise.

Best regards,

Wei Peng

**Comment by GiudGiud** [comment]
@laagesen @amjokisaari

**Reply by laagesen** [reply]
Sorry I'm not knowledgable on crystal plasticity models, I will ask @sapitts or @jiangwen84 if they can help here

**Reply by jiangwen84** [reply]
I do not think we have that capability in MOOSE yet, but it is good to add one. 

@PengWei97 ```ComputeElasticityTensorCP``` and ```PropertyReadFile``` are good starting point.

**Reply by PengWei97** [reply]
Hi @jiangwen84, @laagesen,
Thank your much for your suggestions and your approval of the features I want to add. Below I will try to accomplish my purpose based on the two classes you provided. Of course, I also hope to get your guidance throughout the entire modeling process.
